### PR TITLE
Create redis (Elasticache) instance and a celery worker service

### PR DESCRIPTION
**Pull request by samuelhwilliams**
### Change description
Creates a new Elasticache ("redis") instance in each of our Copilot environments, and a new ECS service called `data-store-celery`, which runs the celery worker and reads tasks from redis.

Updates the deployment pipeline to deploy the celery worker and then the data store API.

- [x] Unit tests and other appropriate tests added or updated
- [x] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

Ticket: https://dluhcdigital.atlassian.net/browse/FPASF-320?atlOrigin=eyJpIjoiZmQyZGMwN2E2ZjNhNDg3OThjY2QzNTVhNjcyZDRlYzIiLCJwIjoiaiJ9

### CODE REVIEW

File: copilot/data-store-celery/addons/data-store-find-data-access.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        Statement:
+          - Sid: S3ObjectActions
+            Effect: Allow
+            Action:
+              - s3:GetObject
+              - s3:PutObject
+              - s3:PutObjectACL
+              - s3:PutObjectTagging
+              - s3:DeleteObject
+              - s3:RestoreObject
```
Comment (gidsg) [inline copilot/data-store-celery/addons/data-store-find-data-access.yml:27]: Can we minimise this permissions set? i.e. I don't think the service should need to delete or restore?

File: copilot/data-store-celery/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+type: Backend Service
+
+# Your service does not allow any traffic.
+
+# Configuration for your containers and service.
+image:
+
+# TODO: re-enable this when secrets are tied up properly Celr
+#  healthcheck:
+#    command: ['CMD-SHELL', 'launcher', 'celery', '-A', 'app.celery_app', 'inspect', 'ping']
```
Comment (gidsg) [inline copilot/data-store-celery/manifest.yml:16]: Can we restore this healthcheck?

Comment (samuelhwilliams) [reply copilot/data-store-celery/manifest.yml:16]: I'm working on it but the values need tweaking a bit.

File: copilot/data-store-celery/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+secrets:
+  NOTIFY_FIND_API_KEY: /copilot/${COPILOT_APPLICATION_NAME}/${COPILOT_ENVIRONMENT_NAME}/secrets/NOTIFY_FIND_API_KEY
+  REDIS_URL:
+    from_cfn: ${COPILOT_APPLICATION_NAME}-${COPILOT_ENVIRONMENT_NAME}-DataStoreRedisUrl
+  DATASTORECLUSTER_SECRET:
+    from_cfn: ${COPILOT_APPLICATION_NAME}-${COPILOT_ENVIRONMENT_NAME}-DataStoreClusterSecret
+
+environments:
+  production:
+    count: 2
```
Comment (gidsg) [inline copilot/data-store-celery/manifest.yml:42]: Maybe we can get away with one instance here as 0 downtime is not actually needed for a worker?

Comment (samuelhwilliams) [reply copilot/data-store-celery/manifest.yml]: Good catch - I forgot to remove this. Have also just set the trace count to 1 in prod, as the number of downloads is so small I'd be curious to get stats on them all (initially at least)...

File: config/envs/aws.py [hunk 4]
```diff
@@ -21,3 +22,10 @@ def get_database_url() -> str:
 @configclass
 class AwsConfig(DefaultConfig):
     SQLALCHEMY_DATABASE_URI = get_database_url()
+
+    # When running tests, don't try to send celery tasks to a worker - instead just run them as if they were
+    # direct synchronous calls. Otherwise we'd have to manage a celery worker/message broker during pytest runs.
+    # Which is overkill for now. 28/06/2024.
```
Comment (gidsg) [inline config/envs/aws.py:28]: This comment doesn't make sense for the production config, can we remove it?

**Review by gidsg — APPROVED** [review]
LGTM *️⃣ 👍

**Pull request merged**